### PR TITLE
Feature/property binding for docket configuration

### PR DESCRIPTION
**Pull request by DavidMueller4242**
Up for discussion: 
* This will be a breaking change if the publishing functionality is used

### CODE REVIEW

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/SpringWolfConfigProperties.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.Map;
+
+@Configuration
+@ConfigurationProperties(prefix = SpringWolfConfigConstants.SPRINGWOLF_CONFIG_PREFIX)
+@ConditionalOnProperty(name = SpringWolfConfigConstants.SPRINGWOLF_ENABLED, matchIfMissing = true)
+@Getter
+@Setter
+public class SpringWolfConfigProperties {
+
+    private boolean enabled = true;
```
Comment (DavidMueller4242) [inline springwolf-core/src/main/java/io/github/stavshamir/springwolf/SpringWolfConfigProperties.java:24]: What would happen if this is false?

File: springwolf-examples/springwolf-kafka-example/src/main/resources/application.properties [hunk 2]
```diff
@@ -1 +1,16 @@
...
+springwolf.docket.info.description=Springwolf example project to demonstrate springwolfs abilities
+springwolf.docket.info.terms-of-service=http://asyncapi.org/terms
+springwolf.docket.info.contact.name=springwolf
+springwolf.docket.info.contact.email=example@example.com
+springwolf.docket.info.contact.url=https://github.com/springwolf/springwolf-core
+springwolf.docket.info.license.name=Apache License 2.0
+springwolf.docket.info.license.url=https://www.apache.org/licenses/LICENSE-2.0.txt
+springwolf.docket.servers.kafka.protocol=kafka
+springwolf.docket.servers.kafka.url=${kafka.bootstrap.servers}
+springwolf.kafka.publishing.enabled=true
```
Comment (timonback) [inline springwolf-examples/springwolf-kafka-example/src/main/resources/application.properties]:
```suggestion
springwolf.binding.kafka.publishing.enabled=true
```

Comment (DavidMueller4242) [reply springwolf-examples/springwolf-kafka-example/src/main/resources/application.properties]: Maybe even plugin?

File: springwolf-examples/springwolf-kafka-example/src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,16 @@
-kafka.bootstrap.servers = ${BOOTSTRAP_SERVER:localhost:29092}
\ No newline at end of file
+kafka.bootstrap.servers = ${BOOTSTRAP_SERVER:localhost:9092}
+spring.application.name=Springwolf example project - Kafka
+springwolf.enabled=true
```
Comment (timonback) [inline springwolf-examples/springwolf-kafka-example/src/main/resources/application.properties:5]:
```suggestion

springwolf.enabled=true
```

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/SpringWolfConfigProperties.java (hunk 1 above)
Comment (sam0r040) [reply springwolf-core/src/main/java/io/github/stavshamir/springwolf/SpringWolfConfigProperties.java:24]: It will now disabled spring wolf completely.

**Comment by timonback** [comment]
Done: Do not use AsyncDocket bean per Spring injection
-> refactor
-> write comment on AsyncApiDocket

**Comment by timonback** [comment]
Done: Introduce AsyncApiDocket interface and Default prefix to implementating class

**Pull request closed without merging**